### PR TITLE
[#164595630] set header buttons min 40x40 

### PR DESCRIPTION
**Pull request by lmorelli986**
This PR aims to set some distance proportions between components in headers as static, in place of a wrong dynamic approach. This is done by the following steps:

* remove dependency of GoBackButton component from BaseScreenComponent
* switch every instance of TouchableHighlight component to Button component in headers, whenever possible
* fix `flex` usage in headers and introduce native-base `noLeft` property
* set dimensions of buttons to match the mockups

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🐞 [#164595630](https://www.pivotaltracker.com/story/show/164595630): I pulsanti presenti nell'header devono avere un area attiva attiva minima di 40x40px

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/components/AbortOnboardingModal.tsx [hunk 1]
```diff
@@ -21,11 +26,11 @@ class AbortOnboardingModal extends React.PureComponent<Props> {
     return (
       <Modal visible={true} onRequestClose={this.props.onClose}>
         <Container>
-          <AppHeader>
-            <Right>
-              <TouchableHighlight onPress={this.props.onClose}>
+          <AppHeader noLeft={true}>
+            <Right style={styles.rightOnly}>
```
Comment (cloudify) [inline ts/components/AbortOnboardingModal.tsx]: I'm not sure about this, why use the `Right` component if it doesn't work here? shouldn't we use a simple `View`? what do you think? - cc @fpersico

File: ts/components/screens/BaseHeader.tsx [hunk 2]
```diff
@@ -34,7 +31,10 @@ export class BaseHeader extends React.PureComponent<Props> {
   public render() {
     const { goBack, headerTitle, onShowHelp, body } = this.props;
     return (
-      <AppHeader primary={this.props.primary}>
+      <AppHeader
+        primary={this.props.primary}
+        noLeft={typeof goBack === "undefined" || goBack === false}
```
Comment (cloudify) [inline ts/components/screens/BaseHeader.tsx]:
```suggestion
        noLeft={goBack !== true}
```

File: ts/screens/messages/MessagesHomeScreen.tsx [hunk 3]
```diff
@@ -75,9 +76,9 @@ const styles = StyleSheet.create({
     // Android shadow
     elevation: 5
   },
-  buttonLeft: {
-    padding: 8,
-    marginLeft: -10
+  ioSearch: {
```
Comment (cloudify) [inline ts/screens/messages/MessagesHomeScreen.tsx]: why we need a style specific for the search button? can we have a more general solution?

File: ts/components/AbortOnboardingModal.tsx (hunk 1 above)
Comment (lmorelli986) [reply ts/components/AbortOnboardingModal.tsx]: Fixed <Right> behaviour. A style property was necessary due to a wrong setting from Headers.ts

File: ts/components/screens/BaseHeader.tsx (hunk 2 above)
Comment (lmorelli986) [reply ts/components/screens/BaseHeader.tsx]: changed.

**Pull request merged**